File: lib/domain/entities/workout_entity.dart
import 'package:equatable/equatable.dart';

class WorkoutEntity extends Equatable {
  final String id;
  final String userId;
  final String title;
  final String type;
  final int durationMinutes;
  final int caloriesBurned;
  final DateTime date;
  final String? notes;
  final DateTime createdAt;
  final DateTime updatedAt;

  const WorkoutEntity({
    required this.id,
    required this.userId,
    required this.title,
    required this.type,
    required this.durationMinutes,
    required this.caloriesBurned,
    required this.date,
    this.notes,
    required this.createdAt,
    required this.updatedAt,
  });

  @override
  List<Object?> get props => [
        id,
        userId,
        title,
        type,
        durationMinutes,
        caloriesBurned,
        date,
        notes,
        createdAt,
        updatedAt,
      ];

  WorkoutEntity copyWith({
    String? id,
    String? userId,
    String? title,
    String? type,
    int? durationMinutes,
    int? caloriesBurned,
    DateTime? date,
    String? notes,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return WorkoutEntity(
      id: id ?? this.id,
      userId: userId ?? this.userId,
      title: title ?? this.title,
      type: type ?? this.type,
      durationMinutes: durationMinutes ?? this.durationMinutes,
      caloriesBurned: caloriesBurned ?? this.caloriesBurned,
      date: date ?? this.date,
      notes: notes ?? this.notes,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}

File: backend/src/workouts/entities/workout.entity.ts
import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';

export enum WorkoutType {
  CARDIO = 'cardio',
  STRENGTH = 'strength',
  FLEXIBILITY = 'flexibility',
  HIIT = 'hiit',
  YOGA = 'yoga',
  SPORTS = 'sports',
  OTHER = 'other',
}

export class Workout {
  @ApiProperty()
  id: string;

  @ApiProperty()
  userId: string;

  @ApiProperty({ example: 'Morning Run' })
  title: string;

  @ApiProperty({ enum: WorkoutType, example: WorkoutType.CARDIO })
  workoutType: WorkoutType;

  @ApiProperty({ example: 30 })
  duration: number;

  @ApiProperty({ example: 250 })
  caloriesBurned: number;

  @ApiPropertyOptional()
  notes?: string;

  @ApiProperty({ type: [String] })
  exercises: string[];

  @ApiProperty()
  createdAt: Date;

  @ApiProperty()
  updatedAt: Date;
}
